src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
```

Comment:
PullProofAI

1. It's recommended to handle errors more specifically in the `githubAuthCallback` and `logout` methods. Instead of checking `error.status`, consider checking for specific error types or conditions to provide more meaningful error responses.
2. Consider logging the errors for better debugging and traceability.

Overall, the error handling can be improved for better clarity and maintainability.

Feel free to reach out if you need further clarification or assistance with the suggested improvements.


<!-- This is an auto-generated comment by PullProofAI -->